Add Spark connector settings (#4577)
opensearch-project/documentation-website: _search-plugins/sql/settings.md

`plugins.sql.cursor.keep_alive` | 1 minute | Configures how long the cursor context is kept open. Cursor contexts are resource-intensive, so we recommend a low value.
`plugins.query.memory_limit` | 85% | Configures the heap memory usage limit for the circuit breaker of the query engine.
`plugins.query.size_limit` | 200 | Sets the default size of index that the query engine fetches from OpenSearch.

## Spark connector settings

> The SQL plugin supports [Apache Spark](https://spark.apache.org/) as a data source. This allows you to run SQL queries in Spark and seamlessly write the resulting data back to OpenSearch. To use Spark, enable the following settings to add Spark as a data source and enable the correct permissions.

Reviewer: The SQL plugin supports Apache Spark as an augmented compute source. Data sources defined in Apache Spark as tables will be consumable from OpenSearch. OpenSearch users will be able to execute SQL queries against external sources from within OpenSearch Dashboards Discover and Observability Logs. To get started, enable the following settings to add Spark as a data source and enable the correct permissions.

Author: Switched it to this: 

---

The SQL plugin supports [Apache Spark](https://spark.apache.org/) as an augmented compute source. When data sources are defined as tables in Apache Spark, OpenSearch can consume those tables. This allows you to run SQL queries against external sources inside OpenSearch Dashboard's [Discover]({{site.url}}{{site.baseurl}}/dashboards/discover/index-discover/) and observability logs. 

To get started, enable the following settings to add Spark as a data source and enable the correct permissions.

Reviewer: That works.

To get started, enable the following settings to add Spark as a data source and enable the correct permissions.

Setting | Description
:--- | :---
`spark.uri` | The identifier for your Spark data source.
`spark.auth.type` | The authorization type used to authenticate into Spark.
`spark.auth.username` | The username for your Spark data source.
`spark.auth.password` | The password for your Spark data source.
`spark.datasource.flint.host` | The host of the Spark data source. Default is `localhost`.
`spark.datasource.flint.port` | The port number for Spark. Default is `9200`.
`spark.datasource.flint.scheme` | The data scheme used in your Spark queries. Valid values are `http` and `https`.
`spark.datasource.flint.auth` | The authorization required to access the Spark data source. Valid values are `false` and `sigv4`.

> `spark.datasource.flint.region` | The region in which your OpenSearch cluster is located. Only use when `auth` is set to `sigv4`. Default value is `us-west-2``.

Reviewer 2: ```suggestion
`spark.datasource.flint.region` | The AWS Region in which your OpenSearch cluster is located. Only use when `auth` is set to `sigv4`. Default value is `us-west-2``.
```

> `spark.datasource.flint.region` | The region in which your OpenSearch cluster is located. Only use when `auth` is set to `sigv4`. Default value is `us-west-2``.

Reviewer 2: When referring to an AWS Region, always use "AWS Region" on first appearance. Capitalized "Region" may be used for subsequent appearances.

> `spark.datasource.flint.write.id_name` | The index name in which the Spark connector writes.

Reviewer 2: ```suggestion
`spark.datasource.flint.write.id_name` | The name of the index to which the Spark connector writes.
```

`spark.datasource.flint.ignore.id_column` | Excludes the `id` column when exporting data in a query. Default is `true`.
`spark.datasource.flint.write.batch_size` | Sets the batch size when writing to a Spark-connected index. Default is `1000`.

> `spark.datasource.flint.write.refresh_policy` | Sets the refresh policy for the Spark connection upon failure for the connector to write data to OpenSearch. Either no refresh (`false`), an immediate refresh (`true`), or a set time to wait `wait_for: X`. Default value is `false`.

Reviewer 2: ```suggestion
`spark.datasource.flint.write.refresh_policy` | Sets the refresh policy for the Spark connection upon failure for the connector to write data to OpenSearch. Either no refresh (`false`), an immediate refresh (`true`), or a set time to wait, `wait_for: X`. Default value is `false`.
```

> `spark.datasource.flint.read.scroll_size` | Sets the amount of results returned by queries ran with Spark. Default is `100`.

Reviewer 3: ```suggestion
`spark.datasource.flint.read.scroll_size` | Sets the amount of results returned by queries run with Spark. Default is `100`.
```

> `spark.datasource.flint.read.scroll_size` | Sets the amount of results returned by queries run with Spark. Default is `100`.

Reviewer 2: ```suggestion
`spark.datasource.flint.read.scroll_size` | Sets the number of results returned by queries run using Spark. Default is `100`.
```
